### ISSUE TITLE
Add RestructuredText support

### ISSUE THREAD

**Issue by thedrow** [dropseed/changerelease#2]
Please add RST support as most Python projects use it to document themselves.

**Comment by davegaeddert** [comment]
It would be awesome to support RST but I don't have much need myself so I can't promise if/when I can finish it out. My initial thought was to use docutils to render the RST to HTML, then do the actual parsing of the HTML because it can be used for the GitHub Release body (RST can't). There are some RST to Markdown converters but that doesn't seem like a very reliable process...

I started down that road but don't have time to play it out right now. PRs are welcome! https://github.com/dropseed/changerelease/tree/rst

**Comment by thedrow** [comment]
Did you try pandoc? It's the goto format converter.

**Comment by davegaeddert** [comment]
I've used pandoc in the past but didn't try it for this yet. My two concerns with it (valid or not) are the added setup/installation complexity of using pandoc as a dependency and potential issues w/ rst features that can't be converted to md. Just seems like using the native tools to convert to HTML might be a more direct path? Might still have to [bleach](https://github.com/mozilla/bleach) the output or something though.

I do think it'd be awesome to support rst "natively" in changerelease but the other thing you could do is run pandoc first, then run changerelease on the converted file?! Haven't tried it yet but that actually feels like the best solution to me...

```yaml
name: changerelease
on:
  workflow_dispatch: {}
  push:
    paths: [CHANGELOG.md]
    branches: [master]
    tags: ["*"]

jobs:
  sync:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    # Convert to markdown first
    # https://github.com/pandoc/pandoc-action-example
    - uses: docker://pandoc/core:2.9
        with:
          args: "CHANGELOG.rst -f rst -t markdown -o CR_CHANGELOG.md"

    - uses: dropseed/changerelease@v1
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
        # optional
        tag_prefix: v
        changelog: CR_CHANGELOG.md
```

@thedrow do you want to try running pandoc first and see how it goes?

**Comment by thedrow** [comment]
Yeah. I'll try it first.

**Comment by davegaeddert** [comment]
Hey @thedrow -- don't know if you had a chance to try this but I did have time to test it out today. The regular expression needed a little tweaking, at least to work on the [celery/celery](https://github.com/celery/celery/blob/master/Changelog.rst) example I tried. I released a new version that I think should work. If you let me know which repo you were wanting to try it on, I'd be happy to open a PR.
